### DISCUSSION TITLE
"One click installer" for linux

### DISCUSSION THREAD

**Discussion by SECRET-GUEST** [lllyasviel/Fooocus#1757] [Show and tell]
[![Python 3.10.12](https://img.shields.io/badge/Python-3.10.12-ff073a?style=flat&logo=python&logoColor=white&colorA=00FFFF)](https://www.python.org/downloads/release/python-31012/)

Hi!

For Linux users with Python 3.10+ (just click on the badge), this script quickly sets up Fooocus for you.

Obviously you first have to install python to make the virtual environment, and git to clone this repo.

###### How to use >>> Paste the script in a `.sh`  file ,make it executable and run it. It checks if Fooocus is already installed; if not, it'll handle the download of dependencies and create a Python environment. Then, you'll be prompted to launch with or without presets - simply enter 1, 2, or 3 for your choice.

You can launch this script via a **shortcut** or any other method when you need to start **Fooocus**.

```bash
#!/bin/bash

# Get the directory of the script
SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"

# Set the working directory to the script's directory
cd "$SCRIPT_DIR"

# Name of the virtual environment
ENV_NAME="py3_VENV_fooocus"
# Absolute path to the Fooocus directory
FOOCUS_DIR="$SCRIPT_DIR/Fooocus"

# Check and create the virtual environment if necessary
if [ ! -d "$ENV_NAME" ]; then
    echo "Creating the virtual environment $ENV_NAME..."
    python3 -m venv $ENV_NAME
fi

# Activate the virtual environment
source $ENV_NAME/bin/activate

# Check if the Fooocus directory exists and contains the file entry_with_update.py
if [ ! -d "$FOOCUS_DIR" ] || [ ! -f "$FOOCUS_DIR/entry_with_update.py" ]; then
    echo "Fooocus installation required..."

    # Clone the Fooocus repository if necessary
    if [ ! -d "$FOOCUS_DIR" ]; then
        echo "Cloning the Fooocus repository..."
        git clone https://github.com/lllyasviel/Fooocus.git $FOOCUS_DIR
    fi

    # Move to the Fooocus directory and install dependencies
    cd $FOOCUS_DIR
    echo "Installing dependencies..."
    pip install -r requirements_versions.txt
else
    # Move to the Fooocus directory
    cd $FOOCUS_DIR
fi

# Menu for launching Fooocus
while true; do
    echo "Choose your preset:"
    echo "1) Anime"
    echo "2) Realistic"
    echo "3) No preset"
    echo ""
    echo -n "Your choice: "
    read choice

    case $choice in
        1)
            echo "Launching with Anime preset..."
            python entry_with_update.py --preset anime
            break
            ;;
        2)
            echo "Launching with Realistic preset..."
            python entry_with_update.py --preset realistic
            break
            ;;
        3)
            echo "Launching Fooocus without preset..."
            python entry_with_update.py
            break
            ;;
        *)
            echo "Invalid choice. Please enter 1, 2, or 3."
            ;;
    esac
done

# Deactivate the virtual environment
deactivate
```

## Big thanks to everyone who has contributed to this project!

![image](https://github.com/lllyasviel/Fooocus/assets/92639080/c3c65f3e-4b5d-4251-bd2b-bacd9b864f1d)

**Comment by sixman9** [comment]
The above script didn't work for me, straight out of the box, on a fresh/very recent install of Ubuntu 22.04.1. To get everything to work I had to:

```bash
sudo apt install git  #I know, pretty standard
sudo apt install python-is-python3  #Gives you the `python` command, linked to the already available `python3` command
sudo apt install python3-pip  #Installs the `pip` installer
pip install pygit2 packaging  #2 packages this Fooocus install script complained about (missing)
```

Hope this is helpful, enjoy.

**Comment by Aerthlyomi** [comment]
Thanks a lot, it worked as a charm.

**Comment by benbois** [comment]
Thanks, works straight away!

**Comment by jVitaliy** [comment]
does not work with python 3.14.x and in Fedora 44, errors after errors, no success

**Reply by SECRET-GUEST** [reply]
Thank you for replying, yes this script is probably outdated, I'm not a Fedora user, and I can't provide you an update for now  but when I'll come back from vaccation I'll try my best at least for the others
